lib/puppet/provider/service/systemd.rb

```diff
@@ -36,10 +36,21 @@ def self.instances
     return []
   end
 
+  # This helper ensures that the enable state cache is always reset
+  # after a systemctl enable operation. A particular service state is not guaranteed
+  # after such an operation, so the cache must be emptied to prevent inconsistencies
+  # in the provider's believed state of the service and the actual state.
+  # @param action [String,Symbol] One of 'enable', 'disable', 'mask' or 'unmask'
+  def systemctl_change_enable(action)
```

Comment:
@MikaelSmith abstracted `enable` operations into a helper - is this what you had in mind? The service tests all still pass.